cmd/tink-worker/client/tink/tink_test.go

```diff
@@ -0,0 +1,151 @@
+package tink
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/packethost/pkg/grpc"
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/tink/protos/workflow"
+	ggrpc "google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+// Self-signed cert we use to test credential generation.
+const selfSignedCert string = `-----BEGIN CERTIFICATE-----
+MIIB0DCCATGgAwIBAgIBATAKBggqhkjOPQQDBDASMRAwDgYDVQQKEwdBY21lIENv
+MB4XDTIxMTIwODIzNTYwMFoXDTIyMDYwNjIzNTYwMFowEjEQMA4GA1UEChMHQWNt
+ZSBDbzCBmzAQBgcqhkjOPQIBBgUrgQQAIwOBhgAEAL71xJa5btFlW1BUaxrLAbsM
+1tgi0UTB6w4CHm+6iwyuP0BZD9eMwSjMzhZ70lmlah0Z5Z8oHFHEFOIYZnIG4O/r
+ATnzswVeVIPuX8uZ6ApsKq5q9uk9ByJIRfLhNqmrpLNSGbdcNEIoI27DIwiQdeT8
+U6WYuWZB8BKBNre2Q/OiAcPNozUwMzAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAww
+CgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAKBggqhkjOPQQDBAOBjAAwgYgCQgE1
+AkrK0PoP1Mvia0jMGYBDwGzBCLjqPpPFuTE0pVJctog/ZXBRUDlSz9BN3PbCFJZs
+F+/RUdtxQsgcBeBNrKk2qwJCAVxHlTzNK+hv7Xr3MXC39TopWJuNEj/1B37FZ8HG
+0qv7ie1O3lcrMXd9evsIp9KLqSgUCyxmN2SS6LKBjaAwzmJd
+-----END CERTIFICATE-----`
+
+type fakeServer struct {
+	workflow.WorkflowServiceServer
+	data map[string]string
+	err  error
+}
+
+const bufSize = 1024 * 1024
+
+func startWorkflowServerAndConnectClient(t *testing.T, name string, server *grpc.Server) workflow.WorkflowServiceClient {
+	t.Helper()
+
+	ctx := context.Background()
+	listener := bufconn.Listen(bufSize)
+	go func() {
+		t.Helper()
+
+		if err := server.Server().Serve(listener); err != nil {
+			t.Error(fmt.Errorf("%s.Serve exited with error: %w", name, err))
```

Comment:
Stopping this test's execution at this point with `t.Fatalf` makes more sense, IMO. The documentation for `Errorf`, it states:

```
Errorf is equivalent to Logf followed by Fail.
```

... and `Fail`'s documentation says:

```
Fail marks the function as having failed but continues execution
```

Reply:
This one is a bit of a special case, because we're in a go routine, and `t.Fatal` can only be called in the same goroutine as the test (I didn't really know that beforehand, the linter told me). ;)